addons/survey/controllers/main.py

```diff
@@ -531,14 +531,17 @@ def _extract_comment_from_answers(self, question, answers):
                     comment = answers['comment'].strip()
                     answers.pop('comment')
                 answers_no_comment = answers
-            else:
-                if not isinstance(answers, list):
-                    answers = [answers]
-                for answer in answers:
-                    if 'comment' in answer:
-                        comment = answer['comment'].strip()
-                    else:
-                        answers_no_comment.append(answer)
+            else :
+                if question.question_type == 'simple_choice' or question.question_type == 'multiple_choice':
+                    if not isinstance(answers, list):
+                        answers = [answers]                 
+                    for answer in answers:
+                        if 'comment' in answer:
```

Comment:
```suggestion
                        if isinstance(answer, dict) and 'comment' in answer:
```

I think all the fix can be replaced by just this check.